frontend/packages/console-app/src/components/volume-snapshot/create-volume-snapshot/create-volume-snapshot.tsx

```diff
@@ -170,11 +147,53 @@ const CreateSnapshotForm = withHandlePromise<SnapshotResourceProps>((props) => {
   }, [namespace, pvcName]);
 
   const [data, loaded, loadError] = useK8sWatchResource<PersistentVolumeClaimKind[]>(resourceWatch);
+  const defaultSnapshotClassAnnotation: string = 'snapshot.storage.kubernetes.io/is-default-class';
+  const isDefaultSnapshotClass = (volumeSnapshotClass: VolumeSnapshotClassKind) =>
+    getAnnotations(volumeSnapshotClass, { defaultSnapshotClassAnnotation: 'false' })[
+      defaultSnapshotClassAnnotation
+    ] === 'true';
+
+  const [scObj, setSCObj] = React.useState<StorageClassResourceKind>(null);
+  const [scName, setSCName] = React.useState('');
+  const [vscObj, vscLoaded, vscErr] = useK8sGet<ListKind<VolumeSnapshotClassKind>>(
+    VolumeSnapshotClassModel,
+  );
+  const [scObjList, scObjListLoaded, scObjListErr] = useK8sGet<ListKind<StorageClassResourceKind>>(
+    StorageClassModel,
+  );
+
+  const scList = scObjListLoaded ? (scObjList.items as StorageClassResourceKind[]) : [];
+  const filter = useCallback(
+    (snapshotClass: VolumeSnapshotClassKind) =>
+      !scObj || !scObj.provisioner ? false : scObj?.provisioner.includes(snapshotClass?.driver),
+    [scObj],
+  );
+
+  const getDefaultItem = useCallback(() => {
+    const vscList = vscLoaded ? (vscObj.items as VolumeSnapshotClassKind[]) : [];
```

Comment:
Move this line outside the call back!
```suggestion
    const vscList = vscLoaded ? (vscObj.items as VolumeSnapshotClassKind[]) : [];
```

Reply:
done